tests/preprocessing/test_data.py

```diff
@@ -575,3 +578,16 @@ def test_transformer_params(self):
         assert pf._transformer.degree == pf.degree
         assert pf._transformer.interaction_only is pf.interaction_only
         assert pf._transformer.include_bias is pf.include_bias
+
+    @pytest.mark.parametrize("daskify", [True, False])
+    def test_df_transform_index(self, daskify):
+        frame = copy(df)
+        if not daskify:
+            frame = frame.compute()
+        frame = frame.sample(frac=1.0)
+        res_df = dpp.PolynomialFeatures(
+            preserve_dataframe=True, degree=1
+        ).fit_transform(frame)
+        if daskify:
+            res_df = res_df.compute()
+        assert_eq_df(res_df.iloc[:, 1:], frame, check_dtype=False)
```

Comment:
Can you explain

1. Why is `check_dtype=False` needed?
2. Why is `.iloc[:, 1:]` needed?

Ideally, we would construct the expected result exactly and compare against that.

Reply:
1. This is a float64 vs float32 thing. So there might be a bigger underlying problem here.
2. The polynomial transformer with degree 1 adds a column of ones to the left of the input data frame. So I can either drop this column of ones from `res_df` or add it to `frame`. Happy to change that.

Reply:
To clarify 1: the input data frame somehow has one float32 column. Haven't figured out yet where that comes from. Since everything gets transformed into numpy arrays under the hood, the float32 column gets upcasted to float64.

Reply:
The first column of `df` gets downcasted in https://github.com/dask/dask-ml/blob/master/tests/preprocessing/test_data.py#L201.

That's why we're ending up with float32 and float64 dtypes in the tests afterwards. I would be inclined to change that test and ensure that `df` does not get mutated. Passing in `df` as a fixture everywhere would probably be better yet.

As for the `PolynomialTransformer`: Because everything gets turned into a numpy array somewhere in sklearn, it's not really a surprise that we're ending up with homogeneous dtypes. I could re-cast everything (just like we're doing with the index) or add a note/comment somewhere that `preserve_dataframe=True` doesn't mean dtypes are preserved.

Reply:
Thanks for looking into it. SGTM.